hasjob/templates/index.html

```diff
@@ -16,6 +17,87 @@
   {%- if location %}<link rel="alternate" type="application/atom+xml" title="Jobs in {{ location['use_title'] }} – {{ config['SITE_TITLE'] }}" href="{{ url_for('feed_by_location', location=location['name']) }}" />{% endif -%}
   {% endblock %}
   {% block description %}{% if location and location.description %}{{ location.description|firstline }}{% else %}{{ super() }}{% endif %}{% endblock %}
+  {% block hasjob_nav %}
+    {%- if showall -%}
+      <div class="header-section">
+        <form id="job-filters" action="{{ url_for('search') }}" role="form">
+          <div class="container">
+           <div class="row">
+              <div class="filters filters-col1 col-xs-12 col-sm-4 col-md-3">
+                <select id="job-filters-location" name="l" multiple="multiple" class="job-filters-select2 notselect" placeholder="Location">
+                  {%- for location in job_locations -%}
+                    <option value="{{location[0]}}" {% if location[0]|string in request.args.getlist('l') %} selected="selected" {% endif %}>{{location[1]}}</option>
```

Comment:
To avoid using `[0]` and `[1]` in the template (they're harder to understand), make the `for` loop `for value, label in job_locations`. Below as well.